src/drive/ducks/files/Toolbar.jsx

```diff
@@ -111,83 +196,9 @@ class Toolbar extends Component {
             className={styles['u-hide--mob']}
           />
         )}
-        <Menu
-          title={t('toolbar.item_more')}
-          disabled={disabled}
-          className={styles['fil-toolbar-menu']}
-          button={<MoreButton />}
-        >
-          {notRootfolder &&
-            !shared.withMe && (
-              <Item>
-                <a
-                  className={styles['fil-action-share']}
-                  onClick={() => this.setState(toggleShowShareModal)}
-                >
-                  {t('toolbar.share')}
-                </a>
-              </Item>
-            )}
-          {shared.withMe && (
-            <Item>
-              <a
-                className={styles['fil-action-share']}
-                onClick={() => this.setState(toggleShowShareModal)}
-              >
-                {t('Files.share.sharedWithMe')}
-              </a>
-            </Item>
-          )}
-          {hasWriteAccess && (
-            <Item>
-              <UploadButton
-                onUpload={files => uploadFiles(files, displayedFolder)}
-                label={t('toolbar.menu_upload')}
-                className={styles['fil-action-upload']}
-              />
-            </Item>
-          )}
-          {actions.addFolder &&
-            hasWriteAccess && (
-              <Item>
-                <a
-                  className={styles['fil-action-newfolder']}
-                  onClick={actions.addFolder}
-                >
-                  {t('toolbar.menu_new_folder')}
-                </a>
-              </Item>
-            )}
-          {notRootfolder && (
-            <Item>
-              <a
-                className={styles['fil-action-download']}
-                onClick={() => downloadAll([displayedFolder])}
-              >
-                {t('toolbar.menu_download_folder')}
-              </a>
-            </Item>
-          )}
-          <Item>
-            <a
-              className={styles['fil-action-select']}
-              onClick={onSelectItemsClick}
-            >
-              {t('toolbar.menu_select')}
-            </a>
-          </Item>
-          {shared.withMe && <hr />}
-          {shared.withMe && (
-            <Item>
-              <a
-                className={classNames(styles['fil-action-delete'])}
-                onClick={() => leaveFolder(displayedFolder)}
-              >
-                {t('toolbar.leave')}
-              </a>
-            </Item>
-          )}
-        </Menu>
+
+        {isMobile ? <BarRight>{MoreMenu}</BarRight> : MoreMenu}
```

Comment:
This is the wrapped part — since `MoreMenu` is now a variable we avoid a lot of duplication.
I almost turned it into a standalone Component but it requires almost every prop in here.

Reply:
I do believe that based-components architecture needs some abstraction to stay organized. I faced the same need with `<FileList />` (which is a collection) and `<File />` (which are items): almost every `FileList.props` are passed to `<File />`, then I am sure it is a matter of abstraction. For instance `<File />` must be a functionnal component and must accept every props needed to be rendered at the lowest level: 
1. display select box or not
1. icon
1. filename
1. etc.

Currently it is a function inside the `FileList` component because it can access `FileList`'s `this.props`. I am not sure it could stay as is, unless items can't be used outside the collection. And maybe that is the very case in my example.

Is it in yours?

Reply:
I would say that no, they can't be used outside of here.

Inside `MoreMenu`, there is a `Menu` and a few `Item` — and [those are already "dumb components"](https://github.com/cozy/cozy-drive/blob/master/src/components/Menu/index.jsx). 
The only thing we could abstract away even more is this very specific configuration of a `Menu`, which isn't going to be used elsewhere. It might improve readability a little, but the resulting Component would not be useful anywhere else.